packages/block-library/src/file/index.js

```diff
@@ -40,20 +39,20 @@ export const settings = {
 		},
 		fileName: {
 			source: 'html',
-			selector: 'a:not([download])',
+			selector: '.wp-block-file__link',
```

Comment:
I'm not sure about all the selector changes - if, for example, the prefix on the classname ever changed (which I think has been discussed) the block wouldn't continue to work in the same way.